File: golang/uPIMulator/src/linker/kernel/instruction/cc/true_false_cc.go
package cc

import (
	"errors"
)

type TrueFalseCc struct {
	condition Condition
}

func (this *TrueFalseCc) Init(condition Condition) {
	conditions := map[Condition]bool{
		TRUE:  true,
		FALSE: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *TrueFalseCc) Condition() Condition {
	return this.condition
}

File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/true_false_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class TrueFalseCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(TrueFalseCC.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return {Condition.TRUE, Condition.FALSE}
